diff --git a/src/components/components/SceneCard/SceneCard.component.jsx b/src/components/components/SceneCard/SceneCard.component.jsx
index c0aad9b5..8836687a 100644
--- a/src/components/components/SceneCard/SceneCard.component.jsx
+++ b/src/components/components/SceneCard/SceneCard.component.jsx
@@ -69,6 +69,7 @@ const SceneCard = ({
   const handleSaveTitle = async () => {
     try {
       const scene = scenesData[editIndex];
+      if (!scene) return;
       await updateSceneIdAndTitle(scene.id, editInputValue);
       const updatedScenes = scenesData.map((s) => {
         if (s.id === scene.id) {
@@ -80,7 +81,7 @@ const SceneCard = ({
       setScenesData(updatedScenes);
       setEditIndex(null);
       AFRAME.scenes[0].components['notify'].message(
-        `New scene title saved: ${editInputValue}`,
+        `New scene title saved, reopen modal to open scene: ${editInputValue}`,
         'success'
       );
     } catch (error) {
@@ -113,13 +114,14 @@ const SceneCard = ({
       handleCancelClick();
     }
   };
+
   return (
     <div className={styles.wrapper}>
       {scenesData?.map((scene, index) => (
         <div key={index} className={styles.card} title={scene.data().title}>
           <div
             className={styles.img}
-            onClick={() => handleSceneClick(scene)}
+            onClick={() => scene.id && handleSceneClick(scene, scene.id)}
             style={{
               backgroundImage: `url(${
                 scene.data().imagePath || ScenePlaceholder
diff --git a/src/components/modals/ScenesModal/ScenesModal.component.jsx b/src/components/modals/ScenesModal/ScenesModal.component.jsx
index 69280dbc..f00eb7aa 100644
--- a/src/components/modals/ScenesModal/ScenesModal.component.jsx
+++ b/src/components/modals/ScenesModal/ScenesModal.component.jsx
@@ -54,13 +54,13 @@ const ScenesModal = ({ isOpen, onClose }) => {
     fetchScenesUser();
   }, [currentUser, isOpen]);
 
-  const handleSceneClick = (scene) => {
+  const handleSceneClick = (scene, currentId) => {
     if (scene.data() && scene.data().data) {
       createElementsForScenesFromJSON(scene.data().data);
       // const sceneId = scene.id;
       window.location.hash = `#/scenes/${scene.id}.json`;
       // this is where we should update sceneid and scenetitle in metadata and toolbar state
-      const sceneId = scene.id;
+      const sceneId = scene.id || currentId;
       const sceneTitle = scene.data().title;
       AFRAME.scenes[0].setAttribute('metadata', 'sceneId', sceneId);
       AFRAME.scenes[0].setAttribute('metadata', 'sceneTitle', sceneTitle);
